src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -0,0 +1,198 @@
+                self.check_len("targets", len(smd_info[valid_info]["vos_ids"]))
+                # Perform additional check for pools uuid and blobs
+                if valid_info == "pools":
+                    if smd_info[valid_info]["uuid"] != self.pool.uuid:
+                        self.fail("Pool uuids don't match: {} {}".format(
+                            smd_info[valid_info]["uuid"], self.pool.uuid))
+                    self.check_len(
+                        "targets", len(smd_info[valid_info]["spdk_blobs"]))
+            else:
+                self.fail("Wanted information not found: {}".format(smd_info))
+
+            if invalid_info in smd_info:
+                self.fail("Found unwanted pool information: {}".format(
+                    smd_info[invalid_info]))
+        else:
+            self.fail("SMD info not found: {}".format(smd_info))
+
+    def check_dev_state_out(self, smd_info, method_name, state):
+        """Check device state output.
+
+        Args:
+            state (str): State expected. e.i. NORMAL | FAULTY
+        """
+        # Parse stdout into a dict check that devices information is displayed.
+        device_state_info = []
+        for info in smd_info:
+            if "Device" in info:
+                uuid = info[info.index("Device") + 1]
+                device_state_info.append(
+                    self.get_dmg_cmd_info(method_name, uuid))
+
+        # Check that the state of each device is NORMAL
+        for dev in device_state_info:
+            if dev[2] != state:
+                self.fail("Found a device in {} state.".format(dev[2]))
+
+    def test_dmg_storage_query_smd_devices(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query smd -d' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_devs,basic
+        """
+        # Get the storage smd infromation, parse and check devices info
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", pools=False)
+        self.check_smd_out(smd_info, "devices", "pools")
+
+        # Check if the number of devices match the config
+        if self.check_len("dbev_list", len(smd_info)):
+            self.fail("Number of devices does not match config: {}".format(
+                len(smd_info)))
+
+    def test_dmg_storage_query_smd_pools(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query smd -p' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_pools,basic
+        """
+        # Create pool and get the storage smd information, then verfify info
+        self.prepare_pool(self.dmg)
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
+        self.check_smd_out(smd_info, "pools", "devices")
+
+        # Destroy pool and get the storage smd information, then verify info
+        self.pool.destroy()
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", devices=False)
```

Comment:
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'devices' in method call
(pylint-no-value-for-parameter) No value for argument 'kwargs' in method call